org.sonarlint.eclipse.cdt/src/org/sonarlint/eclipse/cdt/internal/CdtUtils.java

```diff
@@ -90,8 +91,33 @@ public void configure(IPreAnalysisContext context, IProgressMonitor monitor) {
       .filter(f -> f.getResource() instanceof IFile && fileValidator.test((IFile) f.getResource()))
       .collect(Collectors.toList());
 
+    // This contains the files that will be later removed from the analysis because of the information not being
+    // available from CDT but being crucial for the CFamily analyzer to NOT fail the whole analysis.
+    var removedFilesFromAnalysis = new ArrayList<ISonarLintFile>();
+
     try {
-      var configuredFiles = configureCProject(context, context.getProject(), filesToAnalyze);
+      var configuredFiles = configureCProject(context, context.getProject(), filesToAnalyze, removedFilesFromAnalysis);
+
+      // We have to provide this information back to the analysis when there are files that were removed by the CDT
+      // integration. Due to this relying on extension points, this can only be done via an internal property.
+      if (!removedFilesFromAnalysis.isEmpty()) {
+        var excludedFilesPerUri = removedFilesFromAnalysis.stream()
+          .map(slFile -> slFile.getResource().getLocationURI().toString())
+          .collect(Collectors.toList());
+
+        context.setAnalysisProperty(SonarLintUtils.SONARLINT_ANALYSIS_CDT_EXCLUSION_PROPERY, excludedFilesPerUri);
+
+        logger.error("The following C/C++ files were excluded by the CDT integration as no information could be "
+          + "accessed from the CDT plug-in. This might happen when the CDT plug-in is not yet ready or when the "
```

Comment:
@michael-jabbour-sonarsource there is no official documentation page about it but [one for the Scanner Discovery against the symbols not found errors](https://help.eclipse.org/latest/index.jsp?topic=%2Forg.eclipse.cdt.doc.user%2Fconcepts%2Fcdt_c_scanner_discovery.htm&cp%3D8_2_7_1) and [one about setting the include paths for the indexer](https://help.eclipse.org/latest/index.jsp?topic=%2Forg.eclipse.cdt.doc.user%2Ftasks%2Fcdt_t_sd.htm&cp%3D8_3_7).

I think it might be better to link to the [official documentation](https://help.eclipse.org/latest/index.jsp?topic=%2Forg.eclipse.cdt.doc.user%2Fconcepts%2Fcdt_o_home.htm) as a whole here, WDYT? It contains a lot more information about configuring and tweaking projects ^^

Reply:
Thanks for sharing the resources! For me the most helpful section was:

> 1. Built-in Settings. CDT will try to detect built-in compiler symbols and include paths running the compiler with special options and parse the output of this special run. Most compilers provide such an option to print built-in include paths and symbols. Built-in settings are implied and do not get passed to compiler during regular compilation.
> 2. Build Output Parser (BOP). Another method that CDT employs is to analyze build output of the regular build with Build Output Parser. Often, include paths are supplied to the compiler with -I options, and macros with -D options and BOP will try to find those in the output. That method relies on verbose build output of your build where all these options are actually printed by make.

This helped me understand that, in some cases, CDT inspects the build output to extract the information needed for code highlighting to work (and therefore needs a build to work).

What about having a section in our own docs that says that for C and C++, we rely on the CDT plugin to retrieve the build configuration. We can list a few resources to the official plugin, and hint that building may be required to get this to work. WDYT?

Reply:
I reworked this to spawn a notification instead of just an error message in the log. This also now features the link to the official Eclipse CDT documentation while our Docs team is working on changing it for SonarQube for Eclipse.

For the latter, I already reached out to them, and they are going to create a ticket for it. I will work with them together on the phrasing and the content.

Reply:
Once Docs replied to my mail, I will link the Docs ticket to this Jira ticket ^^